### PR TITLE
Move FabricBot rules to Config-As-Code

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Following [#7442](https://github.com/dotnet/dotnet-api-docs/pull/7442) this PR transfers our FabricBot automation rules to Config-as-Code. Also adds a bit of documentation pointing to the new rule editing workflow.

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

File: docs/infra/automation.md
Comment (eiriktsarpalis) [reply docs/infra/automation.md:7]: For now it needs to be uploaded manually and re-exported.

File: .github/fabricbot.json
Comment (marek-safar) [inline .github/fabricbot.json:1]: Why is this file under .github?

Comment (eiriktsarpalis) [reply .github/fabricbot.json:1]: It's just the convention employed by Fabric Bot

Comment (eiriktsarpalis) [reply .github/fabricbot.json:1]: Cf. https://microsoft.sharepoint.com/teams/FabricBot/SitePages/Config-as-code.aspx

**Review by marek-safar — APPROVED** [review]

**Comment by eiriktsarpalis** [comment]
Since it's a Friday, I'll hold off from merging this until next week. I'll also need to update the FB configuration dump to the most recent state if needed.

File: .github/fabricbot.json
Comment (karelz) [inline .github/fabricbot.json:402]: Nit: Not alphabetically sorted next to other area-System.Net.*

**Review by karelz — APPROVED** [review]

File: .github/fabricbot.json
Comment (eiriktsarpalis) [reply .github/fabricbot.json:402]: This file is the verbatim contents of the exported configuration from the Fabric Bot portal. I thought I'd avoid making changes during the migration to Config-As-Code. I'm planning on iterating with improvements once migration is complete.

**Comment by lewing** [comment]
who has permission to edit the fabric bot config?

**Comment by eiriktsarpalis** [comment]
> who has permission to edit the fabricbot config?

Currently a handful of team members have permissions in the FabricBot portal, but after this is merged anybody should be able to submit a PR.

**Pull request merged**
